src/api_gen.rs

```diff
@@ -1,35 +1,34 @@
 /* Code generated by codegen/main.go. DO NOT EDIT. */
 
 use std::collections::HashMap;
+use std::fmt::{Display, Formatter};
 
-use nanoserde::{DeJson,SerJson};
+use nanoserde::{DeJson, SerJson};
+use serde::{Deserialize, Serialize};
 use urlencoding::encode;
 
-
 #[derive(Debug, Clone)]
 pub enum Authentication {
-  Basic {
-    username: String,
-    password: String
-  },
-  Bearer {
-    token: String
-  }
+    Basic { username: String, password: String },
+    Bearer { token: String },
 }
 
 #[derive(Debug, Clone, PartialEq, Copy)]
 pub enum Method {
-    Post, Get, Put, Delete
+    Post,
+    Get,
+    Put,
+    Delete,
 }
 
 #[derive(Debug, Clone)]
 pub struct RestRequest<Response> {
-  pub authentication: Authentication,
-  pub urlpath: String,
-  pub query_params: String,
-  pub body: String,
-  pub method: Method,
-  _marker: std::marker::PhantomData<Response>
+    pub authentication: Authentication,
```

Comment:
Could you add an .editorconfig or whatever is the idiomatic way to format code in Rust and have it with the old formatting settings so this file is easier to review?